app/models/ad.rb

```diff
@@ -53,6 +60,14 @@ class Ad < ApplicationRecord
   enum category: CATEGORIES
   enum kind: KINDS
 
+  def editable?
+    email.present?
+  end
+
+  def token_valid?(token)
+    token.present? && token == self.token
```

Comment:
Since it's so easy in Rails let's avoid some smartass tring a timing attack and use https://api.rubyonrails.org/classes/ActiveSupport/SecurityUtils.html#method-c-secure_compare instead of regular string compare.

Reply:
Fixed here https://github.com/vlado/earthquake-croatia/pull/110/commits/a7479c3e4d2a42179fccfb4a5c410de066318e8a#diff-a37cbd831db121b0b8405257fa671aa3bcf240c3bcc199eca4d557698296d41dR68